### ISSUE TITLE
SVG files are broken

### ISSUE THREAD

**Issue by maliMirkec** [11ty/eleventy-img#220]
When I try to save the remote image with the Eleventy Image plugin, the SVG file is broken.
I've tried to use Buffer but had no luck.
So I've tried to debug what is going on.

The function `createSvg` in the `src/format-hooks/svg.js` file returns the following Uint8Array data, which is then stored as an SVG file. I don't know how to fix this.

![svg-example](https://github.com/11ty/eleventy-img/assets/2345786/a2be411c-d7b9-4fa5-ac7a-9c49211be3ed)
<img width="1253" alt="svg-js-debug" src="https://github.com/11ty/eleventy-img/assets/2345786/ec74a76a-c202-447b-a57f-f51398e531c0">

**Comment by Aankhen** [comment]
This is a strange bug, as we discussed on Discord, because what you’re getting in the screenshot is the string representation of a Uint8Array. In other words, you’ve somehow got the bytes of the string of the bytes of the image.

**Comment by Aankhen** [comment]
I can confirm that this happens with other SVG images too, like this one: https://upload.wikimedia.org/wikipedia/commons/e/e8/Svg_example3.svg

Here’s a small demonstration:
```
❯❯ cat .\.eleventy.js
const Image = require("@11ty/eleventy-img");

module.exports = function (eleventyConfig) {
  eleventyConfig.addShortcode("image", async function (src, alt, sizes) {
    let metadata = await Image(src, {
      widths: [300, 600],
      formats: [null],
    });

    let imageAttributes = {
      alt,
      sizes,
      loading: "lazy",
      decoding: "async",
    };

    // You bet we throw an error on a missing alt (alt="" works okay)
    return Image.generateHTML(metadata, imageAttributes);
  });
};
❯❯ cat .\test.njk
{% image 'https://images.ctfassets.net/qbmf238cr6te/2ExPY7uYyafazH0IfFnpTD/610bce5faa1598685f2985d2062dcf1f/heyflow-logo.svg', '', '30vw' %}
{% image 'https://upload.wikimedia.org/wikipedia/commons/e/e8/Svg_example3.svg', '', '30vw' %}
❯❯ npx eleventy
[11ty] Writing _site/test/index.html from ./test.njk
[11ty] Wrote 1 file in 0.38 seconds (v2.0.1)
❯❯ la -T img .cache
Mode   Size Date Modified    Date Accessed    Git Name
d----     - 2024-04-15 14:34 2024-04-15 14:34  -N .cache
-a---    87 2024-04-15 14:34 2024-04-15 14:34  -N ├── eleventy-fetch-627a3ab0d5a14cfae88fefbc8d6d06
-a---   477 2024-04-15 14:34 2024-04-15 14:34  -N ├── eleventy-fetch-627a3ab0d5a14cfae88fefbc8d6d06.buffer
-a---    87 2024-04-15 14:34 2024-04-15 14:34  -N ├── eleventy-fetch-732d66e39e1245c8666eeb1d7869e9
-a--- 2.5Ki 2024-04-15 14:34 2024-04-15 14:34  -N └── eleventy-fetch-732d66e39e1245c8666eeb1d7869e9.buffer
d----     - 2024-04-15 14:34 2024-04-15 14:34  -N img
-a--- 1.6Ki 2024-04-15 14:34 2024-04-15 14:34  -N ├── cKJqfkmfxt-467.svg
-a--- 7.5Ki 2024-04-15 14:34 2024-04-15 14:34  -N └── tn8vIuPE6f-1500.svg
❯❯ foreach ($p in (gci .cache | ?{ -not $_.Name.EndsWith(".buffer") })) { cat $p }
[{"627a3ab0d5a14cfae88fefbc8d6d06":"1"},{"cachedAt":1713171842099,"type":"2"},"buffer"]
[{"732d66e39e1245c8666eeb1d7869e9":"1"},{"cachedAt":1713171841858,"type":"2"},"buffer"]
```

**Comment by maliMirkec** [comment]
Does my fix help?

https://github.com/11ty/eleventy-img/pull/221

**Comment by Aankhen** [comment]
It does, but the approach concerns me. There ought to be some way to get the right content from the request.

**Comment by shkarinn** [comment]
> It does, but the approach concerns me. There ought to be some way to get the right content from the request.

But it working, better working approach, than nothing.
In v 4.0 I can't use SVG files.

> Does my fix help?
This commit is working.

**Comment by patrulea** [comment]
I too have this issue with SVG files across all my projects using the later versions of the plugin.

**Comment by peekxc** [comment]
Yeah, this essentially makes the svg format of `eleventy-img` non-functional.

**Comment by zachleat** [comment]
Fixed by #221
